honeybadgermpc/mpc.py

```diff
@@ -186,6 +169,23 @@ def _send(j, o):
 
         return True
 
+    async def _run(self):
+        # Run receive loop as background task, until self.prog finishes
+        # Cancel the background task, even if there's an exception
+        bgtask = asyncio.create_task(self._recvloop())
+        result = asyncio.create_task(self.prog(self, **self.prog_args))
+        await asyncio.wait((bgtask, result), return_when=asyncio.FIRST_COMPLETED)
+
+        if result.done():
+            bgtask.cancel()
+            return result.result()
+        else:
+            logging.info(f'bgtask exception: {bgtask.exception()}')
+            raise bgtask.exception()
+            # FIXME: This code is unreachable and needs to be investigated
```

Comment:
NYC: I think it is safe to remove any code after `raise`

Reply:
I'll leave this for a later refactor